Source/ARTPushActivationStateMachine.m

```diff
@@ -101,7 +101,7 @@ - (void)persist {
 
 - (void)deviceRegistration:(ARTErrorInfo *)error {
     #ifdef TARGET_OS_IOS
-    ARTLocalDevice *local = _rest.device;
+    ARTLocalDevice *local = _rest.device_nosync;
 
     if (![[UIApplication sharedApplication].delegate conformsToProtocol:@protocol(ARTPushRegistererDelegate)]) {
         [NSException raise:@"ARTPushRegistererDelegate must be implemented on AppDelegate" format:@""];
```

Comment:
Wondering: can this fix or impact on this? https://github.com/ably/ably-ios/issues/673

Reply:
@funkyboy If the user is using the `push` branch, then yes.

Reply:
That was a fix for #654.